### ISSUE TITLE
lack of `DeserializeUnsigned` for transaction

### ISSUE THREAD

**Issue by bettybao1209** [neow3j/neow3j#673]
For class `Transaction`, it seems we lack the deserializedUnsigned method to deserialize byte array without witnesses to transaction object.

**Comment by mialbu** [comment]
Hey @bettybao1209, you can use the class `NeoSerializableInterface` to deserialize the serialized byte array of an unsigned transaction like in the code example below. The only difference is that the signer is deserialized into the class `Signer` instead of `AccountSigner`. Adding a witness to the deserialized transaction and then send it works fine. I have tested that and I'm using this approach in an ongoing project myself.

```java
Transaction unsignedTransaction = smartContract.invokeFunction("withdraw", hash160(dev), integer(1))
                .signers(AccountSigner.calledByEntry(owner))
                .getUnsignedTransaction();
byte[] unsignedTxBytes = unsignedTransaction.toArray();
Transaction tx = NeoSerializableInterface.from(unsignedTxBytes, Transaction.class);
```

Try it out and tell us if that's solving your use case or if there is something else missing.

**Comment by gsmachado** [comment]
@bettybao1209 let us know if that's fine for you. 😄

**Comment by bettybao1209** [comment]
![image](https://user-images.githubusercontent.com/12050350/140468738-fe8a6f1f-89a3-4a12-9bc3-d798dc27cad9.png)

![image](https://user-images.githubusercontent.com/12050350/140468709-4c143817-9ad2-407a-be4f-92fcc06b3e40.png)

![image](https://user-images.githubusercontent.com/12050350/140468675-488f839e-c26f-42f6-8105-ee43b167c0a8.png)

I am afraid not, since it also needs the witness to be non-null.

**Comment by bettybao1209** [comment]
> Hey @bettybao1209, you can use the class `NeoSerializableInterface` to deserialize the serialized byte array of an unsigned transaction like in the code example below. The only difference is that the signer is deserialized into the class `Signer` instead of `AccountSigner`. Adding a witness to the deserialized transaction and then send it works fine. I have tested that and I'm using this approach in an ongoing project myself.

Maybe I misunderstood, could you tell me where I maybe wrong?

**Comment by gsmachado** [comment]
@bettybao1209 we merged #675 and I think it would solve this issue. 😄 

We published a version to `v3.13.2-SNAPSHOT`. You can try it out and give us feedback.

I'm closing this issue at the moment. If you test and it's still not working for your use case, please re-open this issue. 👍
